packages/style-engine/src/types.ts

```diff
@@ -0,0 +1,32 @@
+export type Box = {
+	top?: string;
+	bottom?: string;
+	left?: string;
+	right?: string;
+};
+
+export interface Style {
+	spacing?: {
+		padding?: string | Box;
+		margin?: string | Box;
+		lineHeight?: string;
```

Comment:
Tiny nit: should this attribute and all the ones beneath it be in an optional `typography?` object instead of `spacing?`

Reply:
I spotted this as well. I believe you are correct and it will be ironed out as this PR evolves.

Reply:
I've moved this out to a separate `typography` object in https://github.com/WordPress/gutenberg/pull/37978/commits/24f8cb37504fe94efa0c9e094ceede3b8529cb00